### DISCUSSION TITLE
How to get data on Table is relation?

### DISCUSSION THREAD

**Discussion by Jeniphan** [goravel/goravel#254] [Q&A] [answered]
I have models 

type Author struct {
  orm.Model
  Name  string
}

type Book struct {
  orm.Model
  AuthorID   uint
  Name       string
  Date time.Time
  Author     *Author
}

I need get Authors is book have Date is xxx. How i can do? Thanks.

**Comment by hwbrzzl** [comment]
Sorry, I didn't get your meaning clearly, you want to get `Author`, right? You may implement your thoughts through two queries.

**Reply by Jeniphan** [reply]
Yes, I need to get Author. This example data 

Author table.
id name
1 cats
2 dog
3 rabbit

Book table
id name authorID Date
1 aaa 1 01/09/23
2 bb 1 02/09/23
3 cc 2 01/09/23
4 dd 2 02/09/23
5 ee 3 03/09/23
6 ff 3 03/09/23

when where book == 02/09/23


`[
  {
    id: 1,
    name: "cats",
    Book: [
      {
        id:1,
        name: "aaa",
        AuthorId: 1,
        Date: "01/09/23" 
      },
      {
        id:2,
        name: "bb",
        AuthorId: 1,
        Date: "02/09/23" 
      },
    ]
  },
  {
    id: 2,
    name: "dog",
    Book: [
      {
        id:3,
        name: "cc",
        AuthorId: 2,
        Date: "01/09/23" 
      },
      {
        id:4,
        name: "dd",
        AuthorId: 2,
        Date: "02/09/23" 
      },
    ]
  }
]`

**Reply by hwbrzzl** [reply, marked as answer]
You need to get `authorID` from the `Book` table first:

```
var authorIDs []uint
facades.Orm().Query().Model(models.Book{}).Where("Date", "02/09/23").Pluck("authorID", &authorIDs)
```

then:

```
type Author struct {
  orm.Model
  Name string
  Books []*Book // Add this line
}

var authors []models.Author
facades.Orm().Query().Where("id in ?", authorIDs).With("Book").Find(&authors)
```

**Reply by Jeniphan** [reply]
Annnn, Thanks. I think maybe use one query.
